src/components/ebay-icon/component-browser.js

```diff
@@ -1,28 +1,71 @@
 let rootSvg;
+let svgDefs;
+
+function createLinearGradientFromString(data) {
+    // This is needed because if we add linear gradient as a string it will not be rendered.
+    const linearGradient = document.createElementNS(
+        "http://www.w3.org/2000/svg",
+        "linearGradient"
+    );
+    linearGradient.setAttribute("id", data.id);
+    linearGradient.setAttribute("x1", data.x1);
+    linearGradient.setAttribute("y1", data.y1);
+    linearGradient.setAttribute("x2", data.x2);
+    linearGradient.setAttribute("y2", data.y2);
+    linearGradient.setAttribute("gradientUnits", data.gradientUnits);
+    data.children.forEach((stop) => {
+        const newStop = document.createElementNS(
+            "http://www.w3.org/2000/svg",
+            "stop"
+        );
+        newStop.setAttribute("offset", stop.offset);
+        newStop.setAttribute("stop-color", stop["stop-color"]);
+        linearGradient.appendChild(newStop);
+    });
+    return linearGradient;
+}
 
 export default {
     onMount() {
         // Create a hidden svg to store all symbols on startup.
         if (!rootSvg) {
-            rootSvg = document.createElement("svg");
-            const rootDiv = document.createElement("div");
-            rootDiv.hidden = true;
-            document.body.insertBefore(rootDiv, document.body.firstChild);
-            rootDiv.appendChild(rootSvg);
+            rootSvg = document.createElementNS(
+                "http://www.w3.org/2000/svg",
+                "svg"
+            );
+            // Apply "hidden styles" to the svg. We don't use display none because then the svg will not be rendered.
+            rootSvg.style.position = "absolute";
+            rootSvg.style.width = "0";
+            rootSvg.style.height = "0";
+
+            document.body.insertBefore(rootSvg, document.body.firstChild);
+            svgDefs = document.createElementNS(
+                "http://www.w3.org/2000/svg",
+                "defs"
+            );
+            rootSvg.appendChild(svgDefs);
         }
 
         // If there were any symbols rendered then we move them to the svg above after rendering them.
         const defs = this.getEl("defs");
 
         if (defs) {
+            let defItem;
             if (this.input && this.input._themes) {
                 defs.innerHTML = this.input._themes();
             }
+            if (this.input && this.input._def) {
+                defItem = createLinearGradientFromString(this.input._def());
```

Comment:
Can we create a logical fork here to make it a bit more scalable right off the bat? Check inside `defs` with a `switch` statement and crate the linear gradient when `linearGradient` is inside? We're probably going to have more gradient types upcoming. It will be easier to manage this if the fork is already set up.

Reply:
As I mentioned in the description
> For now it seems like these attributes are the only ones used. We can revisit to make it more generic
Yes, when we get more types we can add that case and make the attributes setting more generic. I think what we will end up doing is putting the attributes in it's own key and iterating through them, like:
```
{ 
   type: "linearGradient",
   attributes: { id: 'id', x1: 'x1" etc}
   children: { //etc }
}
```
This won't be a breaking change and will make it much more generic. So when we do get more types, Im happy to add that, but since this is a limited change we can leave it for now. The main rationale is I want to see what the other types we get are and then figure it out to make it generic then. That way we won't be guessing and we will see what we need to support fully.

Reply:
Yea, I saw that in the PR. I think it'd be best to bake that in ahead of time, but if you want to iterate on this, I guess that's fine too.